app/code/Magento/Sales/Test/Unit/Controller/Guest/ReorderTest.php

```diff
@@ -0,0 +1,141 @@
+use Magento\Framework\Registry;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager as ObjectManagerHelper;
+use Magento\Sales\Controller\Guest\OrderLoader;
+use Magento\Sales\Controller\Guest\Reorder;
+use Magento\Sales\Helper\Reorder as ReorderHelper;
+use Magento\Sales\Model\Order;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test class for Reorder
+ *
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class ReorderTest extends TestCase
+{
+    /**
+     * Stub Order Id
+     */
+    private const STUB_ORDER_ID = 1;
+
+    /**
+     * @var Reorder
+     */
+    private $reorder;
+
+    /**
+     * @var Registry|MockObject
+     */
+    private $registryMock;
+
+    /**
+     * @var OrderLoader|MockObject
+     */
+    private $orderLoaderMock;
+
+    /**
+     * @var RequestInterface|MockObject
+     */
+    private $requestMock;
+
+    /**
+     * @var RedirectFactory|MockObject
+     */
+    private $resultRedirectFactoryMock;
+
+    /**
+     * @var ReorderHelper|MockObject
+     */
+    private $reorderHelperMock;
+
+    /**
+     * @var MessageManagerInterface|MockObject
+     */
+    private $messageManagerMock;
+
+    /**
+     * Setup environment for test
+     */
+    protected function setUp()
+    {
+        $contextMock = $this->createMock(Context::class);
+        $this->registryMock = $this->createMock(Registry::class);
+        $this->orderLoaderMock = $this->createMock(OrderLoader::class);
+        $this->requestMock = $this->createMock(RequestInterface::class);
+        $this->resultRedirectFactoryMock = $this->createMock(RedirectFactory::class);
+        $this->reorderHelperMock = $this->createMock(ReorderHelper::class);
+        $this->messageManagerMock = $this->createMock(MessageManagerInterface::class);
+
+        $contextMock->expects($this->once())->method('getRequest')->willReturn($this->requestMock);
+        $contextMock->expects($this->once())->method('getResultRedirectFactory')
+            ->willReturn($this->resultRedirectFactoryMock);
+        $contextMock->expects($this->once())->method('getMessageManager')
+            ->willReturn($this->messageManagerMock);
+
+        $objectManagerMock = $this->createMock(ObjectManagerInterface::class);
+        $objectManagerMock->expects($this->once())->method('get')
+            ->with(ReorderHelper::class)
+            ->willReturn($this->reorderHelperMock);
+
+        ObjectManager::setInstance($objectManagerMock);
+
+        $objectManager = new ObjectManagerHelper($this);
+        $this->reorder = $objectManager->getObject(
+            Reorder::class,
+            [
+                'context' => $contextMock,
+                'orderLoader' => $this->orderLoaderMock,
+                'registry' => $this->registryMock
+            ]
+        );
+    }
+
+    /**
+     * Test execute() with the reorder is not allowed
+     */
+    public function testExecuteWithReorderIsNotAllowed()
```

Comment:
It's just recommendation to build test using the pattern:
* Given
* When
* Then

In your case:
```php
// Given
        $orderMock = $this->createMock(Order::class);
        $orderMock->method('getId')->willReturn(self::STUB_ORDER_ID);

        $this->orderLoaderMock->method('load')
            ->with($this->requestMock)
            ->willReturn($this->resultRedirectFactoryMock);

        $this->registryMock->expects($this->once())->method('registry')
            ->with('current_order')
            ->willReturn($orderMock);

// When
        $this->reorderHelperMock->method('canReorder')->with(self::STUB_ORDER_ID)
            ->willReturn(false);

// Then
        $resultRedirectMock = $this->createMock(Redirect::class);
        $this->resultRedirectFactoryMock->expects($this->once())->method('create')->willReturn($resultRedirectMock);
        $this->reorderHelperMock->method('canReorder')->with(self::STUB_ORDER_ID)
            ->willReturn(false);

        /** Expected Error Message */
        $this->messageManagerMock->expects($this->once())
            ->method('addErrorMessage')
            ->with('Reorder is not available.')->willReturnSelf();
        $resultRedirectMock->expects($this->once())
            ->method('setPath')
            ->with('checkout/cart')->willReturnSelf();

        /** Assert result */
        $this->assertEquals($resultRedirectMock, $this->reorder->execute());
```

I hope I haven't lost anything :-)

With such structure, it's easier to read tests, because you have clearly separated stuff that is responsible for preparing the context for test `Given`, the condition `When` and the assertions `Then`